packages/hash/frontend/src/components/layout/PageSidebar/AccountPageList/PageTreeItem.tsx

```diff
@@ -3,27 +3,32 @@ import TreeItem, {
   TreeItemProps,
   useTreeItem,
   TreeItemContentProps,
+  treeItemClasses,
 } from "@mui/lab/TreeItem";
 // import clsx from "clsx";
-import { Box, Typography } from "@mui/material";
-import { faChevronRight, faFile } from "@fortawesome/free-solid-svg-icons";
+import { Box, Tooltip, Typography } from "@mui/material";
+import { usePopupState, bindTrigger } from "material-ui-popup-state/hooks";
+import {
+  faChevronRight,
+  faEllipsis,
+  faFile,
+} from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "../../../icons";
 import { IconButton } from "../../../IconButton";
+import { Link } from "../../../Link";
+import { PageMenu } from "./PageMenu";
 
-type CustomContentProps = TreeItemContentProps & { hasChildren?: boolean };
+// tweaked the example at https://mui.com/components/tree-view/#IconExpansionTreeView.tsx
+const CustomContent = React.forwardRef((props: TreeItemContentProps, ref) => {
+  const { label, nodeId, expandable, url, depth } = props;
+  const [hovered, setHovered] = React.useState(false);
```

Comment:
I think both approaches work. I used it here so that the styles are less convoluted. `useTreeItem` provides `selected` and `focused` in a similar manner. If you insist we shouldn't use this, then I'll address in #440

Reply:
You're right both approaches work, but it's my view that it is best practice to use CSS selectors over custom JS variables because:
- we don't have to maintain the logic that ensures the state is kept up-to-date
- CSS selectors are more performant in the browser than JS function calls + react state re-render
- CSS selectors consume less memory than JS functions/variables/`useState` to make the equivalent logic work.

I think the main reason `useTreeItem` provides `focused` is because the react component hierarchy is quite complex for trees, which is why storing it in a shared context can be justified.

However as far as I can tell from the places you've introduces `hovered` state there is no such complexity that needs to be accounted for. If this is too time consuming to address I suppose we can add a `@todo` to refactor this down the line, I just want to avoid this approach being needlessly introduced throughout the codebase.

Reply:
Duly noted. Will fix in #440

Reply:
Thanks for hearing me out :)